fairscale/optim/oss.py

```diff
@@ -367,3 +369,14 @@ def _broadcast_state_dict(self) -> None:
                 # Discard this tensor/rank, broadcast necessary for syncing
                 logging.debug("Discarding broadcast from rank %s", rank)
                 broadcast_object(empty_buffer, src_rank=rank, group=self.group, dist_device=self._device)
+
+    def _free_other_grads(self) -> None:
```

Comment:
nice find. perhaps we can do this while doing reduce with other rank? I.e. we can immediately free the grad after they are used? feel free to defer to a separate change.

Reply:
in sharded_ddp you mean ? Yes I think that there are tons of optimizations to be done there, for now it's pretty slow, which is to be expected. Here I just wanted to fix it in the general case, when step() is called with the normal DDP